Fix profileManager.test.ts: Test implementation details instead of behavior

## Problem

The release workflow failed because `profileManager.test.ts` test `should load profile and import to SettingsService` is testing implementation details (mock call arguments) rather than observable behavior.

### Failing Test
```
FAIL  src/config/profileManager.test.ts > ProfileManager > load method with SettingsService > should load profile and import to SettingsService
AssertionError: expected "spy" to be called with arguments: [ { defaultProvider: 'openai', …(2) } ]
```

### Root Cause
The test explicitly asserts that `importFromProfile` is called with a specific object shape including `baseUrl: 'https://api.openai.com/v1'`:

```typescript
expect(mockSettingsService.importFromProfile).toHaveBeenCalledWith({
  defaultProvider: 'openai',
  providers: {
    openai: {
      // ...
      baseUrl: 'https://api.openai.com/v1',  // <-- This changed to 'base-url' in PR #1491
    },
  },
  // ...
});
```

This violates `dev-docs/RULES.md`:

> ### What NOT to Test
> ❌ Implementation details  
> ❌ Mock interactions

> ### Mistake 1: Testing Implementation
> ```typescript
> // ❌ BAD
> it('should call database.find', () => {
>   const mockDb = { find: jest.fn() };
>   service.getUser('123');
>   expect(mockDb.find).toHaveBeenCalledWith('123');
> });
> ```

### Proposed Fix
1. Refactor the test to verify **behavior** (e.g., that the profile is successfully loaded and applied)
2. Remove assertions on exact mock call arguments
3. Or if integration with SettingsService is important, test the actual outcome rather than the internal call signature

### Related
- PR #1491 standardized settings keys to kebab-case (`base-url`)
- This exposed the fragile test that depended on internal key naming

--- a/packages/core/src/config/profileManager.test.ts
+++ b/packages/core/src/config/profileManager.test.ts
@@ -143,11 +143,6 @@ describe('ProfileManager', () => {
   });
 
   describe('save method with SettingsService', () => {
-    beforeEach(() => {
-      // SettingsService is already mocked in the main beforeEach
-      profileManager = new ProfileManager();
-    });
-
     it('should export from SettingsService and save profile', async () => {
       const settingsData = {
         defaultProvider: 'openai',
@@ -180,9 +175,6 @@ describe('ProfileManager', () => {
     });
 
     it('should work when SettingsService is always available', async () => {
-      // In the new architecture, SettingsService is always available
-      const manager = new ProfileManager();
-
       mockSettingsService.exportForProfile.mockResolvedValue({
         defaultProvider: 'openai',
         providers: {
@@ -200,7 +192,7 @@ describe('ProfileManager', () => {
       mockFs.writeFile.mockResolvedValue();
 
       await expect(
-        manager.save(
+        profileManager.save(
           'test-profile',
           mockSettingsService as unknown as SettingsService,
         ),
@@ -208,7 +200,6 @@ describe('ProfileManager', () => {
     });
 
     it('should persist tool enablement lists from settings service', async () => {
-      const manager = new ProfileManager();
       const payloadCapture: { value?: unknown } = {};
 
       mockSettingsService.exportForProfile.mockResolvedValue({
@@ -229,7 +220,7 @@ describe('ProfileManager', () => {
         payloadCapture.value = JSON.parse(data);
       });
 
-      await manager.save(
+      await profileManager.save(
         'tool-profile',
         mockSettingsService as unknown as SettingsService,
       );
@@ -246,7 +237,6 @@ describe('ProfileManager', () => {
     });
 
     it('should persist toolFormat from settings service to ephemeralSettings', async () => {
-      const manager = new ProfileManager();
       const payloadCapture: { value?: unknown } = {};
 
       mockSettingsService.exportForProfile.mockResolvedValue({
@@ -264,7 +254,7 @@ describe('ProfileManager', () => {
         payloadCapture.value = JSON.parse(data);
       });
 
-      await manager.save(
+      await profileManager.save(
         'kimi-profile',
         mockSettingsService as unknown as SettingsService,
       );
@@ -277,11 +267,6 @@ describe('ProfileManager', () => {
   });
 
   describe('load method with SettingsService', () => {
-    beforeEach(() => {
-      // SettingsService is already mocked in the main beforeEach
-      profileManager = new ProfileManager();
-    });
-
     it('should load profile and invoke importFromProfile', async () => {
       const profileJson = JSON.stringify(testProfile);
       mockFs.readFile.mockResolvedValue(profileJson);

